model-optimizer/extensions/middle/StridedSliceNormalizer.py

```diff
@@ -0,0 +1,251 @@
+            StridedSliceNormalizer.extend_inputs(node, num_insertions)
+
+        if num_insertions > 0:
+            # insert blank values for ellipsis unrolling and extending
+            for mask_name in StridedSlice.get_mask_names():
+                node[mask_name] = np.insert(node[mask_name], insertion_start_idx, [0] * num_insertions).astype(int)
+
+    @staticmethod
+    def unroll_ellipsis_for_inputs(graph: Graph, node: Node, ellipsis_start: int, num_insertions: int):
+        node_name = node.soft_get('name', node.id)
+
+        for i, input_name in [(1, 'begin'), (2, 'end'), (3, 'strides')]:
+            blank_values_arr = np.zeros(num_insertions) if input_name != 'strides' else np.ones(num_insertions)
+            blank_values_node = Const(graph, {'name': node_name + '/const_to_unroll_{}_ellipsis'.format(input_name),
+                                              'value': int64_array(blank_values_arr)}).create_node()
+
+            if input_name == 'strides' and node.in_port(3).disconnected():
+                continue  # no need to extend strides if they are not connected
+
+            concat_in_ports_count = 3 if ellipsis_start != 0 else 2
+            concat = Concat(graph, {'axis': 0, 'name': node_name + '/concat_{}'.format(input_name),
+                                    'in_ports_count': concat_in_ports_count}).create_node()
+
+            if ellipsis_start != 0:
+                split = create_op_with_const_inputs(graph, VariadicSplit, {1: int64_array(0),
+                                                                           2: int64_array([ellipsis_start, -1])},
+                                                    {'name': node_name + '/split_for_{}_ellipsis'.format(input_name),
+                                                     'out_ports_count': 2})
+                node.in_port(i).get_connection().set_destination(split.in_port(0))
+
+                concat.in_port(0).connect(split.out_port(0))
+                concat.in_port(1).connect(blank_values_node.out_port(0))
+                concat.in_port(2).connect(split.out_port(1))
+            else:
+                concat.in_port(0).connect(blank_values_node.out_port(0))
+                node.in_port(i).get_connection().set_destination(concat.in_port(1))
+
+            concat.out_port(0).get_connection().set_destination(node.in_port(i))
+
+    @staticmethod
+    def extend_inputs(node: Node, num_insertions: int):
+        graph = node.graph
+        node_name = node.soft_get('name', node.id)
+
+        for i, input_name in [(1, 'begin'), (2, 'end'), (3, 'strides')]:
+            blank_values_arr = np.zeros(num_insertions) if input_name != 'strides' else np.ones(num_insertions)
+            blank_values_node = Const(graph, {'name': node_name + '/extend_{}_const'.format(input_name),
+                                              'value': int64_array(blank_values_arr)}).create_node()
+
+            if input_name == 'strides' and node.in_port(3).disconnected():
```

Comment:
May be more convenient: `if i == 3 and node.in_port(3).disconnected():`

Reply:
Agreed